model/courseAllocationModel.js

```diff
@@ -14,24 +13,46 @@ const courseAllocationModel = new mongoose.Schema({
         batch:{
             type: mongoose.Schema.ObjectId,
             ref: 'Batch',
-        },
-        type: Object
+        }
     },
-    course : {
-        teacherId:{
+    course : [{
+        teacher:{
             type: mongoose.Schema.ObjectId,
             ref: 'Teacher',
         },
-        section: {
+        section:[{
             type: mongoose.Schema.ObjectId,
             ref: 'Section',
-        },
-        courseId :{
+        }],
+        course :{
             type: mongoose.Schema.ObjectId,
             ref: 'Course',
-        },
-        type: Array
-    }
+        }
+    }]
+    
 })
+// Query middleware
```

Comment:
use query middleware.The population is the process of automatically replacing the specified paths in the document with the document(s) from other collection(s). We may populate a single document, multiple documents, a plain object, multiple plain objects, or all objects returned from a query